cranelift/control/src/chaos.rs

```diff
@@ -62,4 +62,17 @@ impl ControlPlane {
         self.tmp.copy_from_slice(rest);
         std::mem::swap(&mut self.data, &mut self.tmp);
     }
+
+    /// Returns a new iterator over the same items as the input iterator in
+    /// a pseudo-random order if the control plane was constructed with
+    /// `arbitrary`.
+    ///
+    /// The default value, an iterator with an unchanged order, will always
+    /// be returned if the pseudo-random data is exhausted or the control
+    /// plane was constructed with `default`.
+    pub fn get_permutation<T>(&mut self, iter: impl Iterator<Item = T>) -> impl Iterator<Item = T> {
```

Comment:
Let's call this `shuffled(iter)` or similar, perhaps? Otherwise we're using two different technical words ("shuffle" and "permutation") for method names, which seems unnecessary to me.